### DISCUSSION TITLE
Safari does not update the page automatically

### DISCUSSION THREAD

**Discussion by leofeyer** [community/community#106593] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

When I open a pull request in both Safari 17.0 and Chrome 118.0 and then merge this pull request on the command line with `gh pr merge`, the status of the pull request in Chrome automatically changes to "merged". However, nothing is updated in Safari until I reload the page.

Clearing their website data for github.com fixes the problem temporarily, but it comes back pretty quickly after that. While debugging this, I noticed the `socket-worker.js` file is no longer loaded when the problem occurs.

https://github.com/community/community/assets/1192057/a76cdc00-757c-4b27-b1b3-784cce591cce

Does anyone have the same problem and know how to fix it?

**Comment by sapalazz** [comment]
Hey! This has never happened to me, however, sometimes our pages have struggled to update and some of the things that have worked have been:

- Clearing the cache, entering via incognito mode or clearing the cookies.
- Uninstalling browser extensions that could also interfere.
- Using other browsers (literally, try to replicate it in any browser you can think of... Safari, Chrome, Firefox, Edge, etc.) to see if it's related to any of these browsers or if it's a problem that sticks to all.

Other things I can think of is resetting the settings of those browsers that are giving you problems. But that's only off the top of my head.

Sorry I can't be of any more help, really hope you can get this fixed soon.

**Comment by ChristopheLav** [comment]
I encounter this issue too since a while (months) and I don't find any valuable workaround. This issue seems to be specific to Safari.

The issue is not only related to the Pull Requests hub, also present in the Actions hub. I need to refresh the page to see the real state of the action run. If I completely restart Safari, this is working for some times again then the issue reappear.

This is clearly a bad user experience specific to Safari due to this issue. I need to open a Chrome instance for GitHub only...

Note 1: clear cache or disabling extensions don't help.

Note 2: there is a WebSocket log when the issue occurred `WebSocket connection to 'wss://pipelinesghubeus2.actions.githubusercontent.com/[....]' failed: WebSocket is closed due to suspension.`, maybe related ?

**Reply by mosherc** [reply]
This also happens to me a lot on Safari on PRs. I will get lots of stale information about the merge status of the PR, whether it needs updating with main or not, and it causes weird errors when I click buttons because the state shown on the page doesn't match what their backend expects.

**Comment by JuliaMoon1** [comment]
I got the same issue

**Comment by leofeyer** [comment]
FYI, I tried disabling the back/forward cache as described [here](https://discussions.apple.com/thread/4161818?answerId=19252220022&sortBy=best#19252220022) to see if that was causing the WebSocket suspension, but unfortunately it didn't help.

<img width="806" alt="" src="https://github.com/community/community/assets/1192057/b939e8dc-47b2-4550-8713-8c38325c1930">

I stumbled across it after reading https://stackoverflow.com/a/46457407.

**Comment by leofeyer** [comment]
This seems to happen every time the problem occurs, too:

<img width="296" alt="" src="https://github.com/community/community/assets/1192057/e19e4497-fdeb-42b5-8662-882f1038c3aa">

Does anyone know what this is?

**Comment by PatTheMav** [comment]
Can confirm the same happening to me recently. The only fix I found is to log out and back in again. Maybe related to using a passkey for login?

**Reply by PatTheMav** [reply]
@leofeyer There are also numerous reports of users in the discussion on the updated actions output about it not working in Safari at all, maybe chime in there as well: https://github.com/orgs/community/discussions/89879

**Reply by leofeyer** [reply]
Thanks for the heads-up, but my problem is not related to actions alone. When the error occurs, it affects all of github.com; actions as well as pull requests and notifications – basically everything that is usually updated dynamically.

**Comment by samdoran** [comment]
This has been driving me crazy for several months now. I have not seen the "pages not responding" popup but I do see this error in the web inspector console:
```
Beacon API cannot load https://api.github.com/_private/browser/stats. Reached maximum amount of queued data of 64Kb for keepalive requests
```

The only temporary workaround I have found is deleting all GitHub website data and logging in again. But it comes back quite quickly.

I'm not sure if this is related to Safari Profiles since I have only seen the failure to refresh happen in Profiles other than the default. That could just be coincidence, though.

**Reply by paramountsky9990** [reply]
ok

**Comment by ChristopheLav** [comment]
Hi all, recently the current GitHub Actions Streaming Logs in Beta put more focus on this Safari issue. Today, GitHub has indicated a fix has been released for the issue. After a few tests, the issues describe in this thread _seems_ also fixed 🎉

https://github.com/orgs/community/discussions/89879#discussioncomment-9192239

**Reply by paramountsky9990** [reply]
ok

**Reply by samdoran** [reply]
That would be awesome. I'll see if it's any better today.

**Comment by leofeyer** [comment]
The problem no longer seems to occur since yesterday. 🥳
